### PR TITLE
Updated Matrix_Expo.mdx with java code

### PR DESCRIPTION
**Pull request by 19WH1A0524**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

**Comment by SansPapyrus683** [comment]
image
read them again

File: content/5_Plat/Matrix_Expo.mdx [hunk 1]
```diff
@@ -129,7 +129,64 @@ int main() {
...
+<JavaSection>
+
+```java
+
+import java.util.*;
+import java.lang.*;
+import java.io.*;
+
+class Main{
+public static int MOD = 1000000007;
```
Comment (ChuangSheep) [inline content/5_Plat/Matrix_Expo.mdx:141]:
```suggestion
public final static int MOD = 1000000007;
```

File: content/5_Plat/Matrix_Expo.mdx [hunk 2]
```diff
@@ -129,7 +129,64 @@ int main() {
...
+public static long[][] m = {{1,1},{1,0}};
+public static long[][] m1 = {{1,0},{0,1}};
+public static long[][] mul(long[][] A,long[][] B,int MOD) {
```
Comment (ChuangSheep) [inline content/5_Plat/Matrix_Expo.mdx:144]:
```suggestion
public static long[][] mul(long[][] A, long[][] B, int MOD) {
```
spacing

File: content/5_Plat/Matrix_Expo.mdx [hunk 3]
```diff
@@ -129,7 +129,64 @@ int main() {
...
+    }
+    long[][] half = power(matrix, n / 2, MOD);
+    long[][] square = mul(half, half, MOD);
+    if (n % 2 == 0) {
+        return square;
+    }
+    return mul(matrix, square, MOD);
+}
+ 
+
```
Comment (ChuangSheep) [inline content/5_Plat/Matrix_Expo.mdx:171]:
```suggestion
```

File: content/5_Plat/Matrix_Expo.mdx [hunk 4]
```diff
@@ -129,7 +129,64 @@ int main() {
...
+    public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
+        long n = sc.nextLong();  
+        if (n == 0) {
+            System.out.println(0);
+        } else if(n > 2) {
```
Comment (ChuangSheep) [inline content/5_Plat/Matrix_Expo.mdx:177]:
```suggestion
        } else if (n > 2) {
```
spacing

File: content/5_Plat/Matrix_Expo.mdx [hunk 5]
```diff
@@ -129,7 +129,64 @@ int main() {
...
+            long[][] o = power(m, n-1, MOD);
+            System.out.println(o[0][0]%MOD);
```
Comment (ChuangSheep) [inline content/5_Plat/Matrix_Expo.mdx:179]:
```suggestion
            System.out.println(o[0][0] % MOD);
```
spacing

File: content/5_Plat/Matrix_Expo.mdx [hunk 6]
```diff
@@ -129,7 +129,64 @@ int main() {
...
+        } else {
+            System.out.println(1);
+        }
+     
+    
+    }
```
Comment (ChuangSheep) [inline content/5_Plat/Matrix_Expo.mdx:185]:
```suggestion
    }
```

**Review by ChuangSheep — COMMENTED** [review]
and 
1. as mentioned, please read the code conventions again and **use tab to indent**. Also remember to fix your indentation for ll. 141-169. 
2. spacing

File: content/5_Plat/Matrix_Expo.mdx [hunk 7]
```diff
@@ -129,6 +129,60 @@ int main() {
...
+
+```java
+
+import java.util.*;
+import java.io.*;
+
+class FibonacciNumbers {
+	public final static int MOD = 1000000007;
+	public static long[][] m = {{1,1},{1,0}};
+	public static long[][] m1 = {{1,0},{0,1}};
```
Comment (ChuangSheep) [inline content/5_Plat/Matrix_Expo.mdx:142]:
```suggestion
	public static long[][] m = {{1, 1}, {1, 0}};
	public static long[][] m1 = {{1, 0}, {0, 1}};
```

File: content/5_Plat/Matrix_Expo.mdx [hunk 8]
```diff
@@ -129,6 +129,60 @@ int main() {
...
+        		return matrix;
+    		}
+		long[][] half = power(matrix, n / 2, MOD);
+		long[][] square = mul(half, half, MOD);
+		if (n % 2 == 0) {
+			return square;
+		}
+		return mul(matrix, square, MOD);
+	}
+	public static void main(String[] args) {
```
Comment (ChuangSheep) [inline content/5_Plat/Matrix_Expo.mdx:169]:
```suggestion
	}
	
	public static void main(String[] args) {
```

File: content/5_Plat/Matrix_Expo.mdx [hunk 9]
```diff
@@ -129,6 +129,60 @@ int main() {
...
+		Scanner sc = new Scanner(System.in);
+        	long n = sc.nextLong();  
+        	if (n == 0) {
+            		System.out.println(0);
+        	} 
+		else if (n > 2) {
+            		long[][] o = power(m, n-1, MOD);
```
Comment (ChuangSheep) [inline content/5_Plat/Matrix_Expo.mdx:176]:
```suggestion
            		long[][] o = power(m, n - 1, MOD);
```

**Review by ChuangSheep — APPROVED** [review]
everything elses looks fine 🎉

File: content/5_Plat/Matrix_Expo.mdx [hunk 10]
```diff
@@ -129,6 +129,61 @@ int main() {
...
+```java
+
+import java.util.*;
+import java.io.*;
+
+class FibonacciNumbers {
+	public final static int MOD = 1000000007;
+	public static long[][] m = {{1, 1}, {1, 0}};
+	public static long[][] m1 = {{1, 0}, {0, 1}};
+	public static long[][] mul(long[][] A, long[][] B, int MOD) {
```
Comment (SansPapyrus683) [inline content/5_Plat/Matrix_Expo.mdx:144]: rename to `a`, `b`, and `mod`

Comment (SansPapyrus683) [reply content/5_Plat/Matrix_Expo.mdx:144]: also comment explaining what this does

File: content/5_Plat/Matrix_Expo.mdx [hunk 11]
```diff
@@ -129,6 +129,61 @@ int main() {
...
+
+```java
+
+import java.util.*;
+import java.io.*;
+
+class FibonacciNumbers {
+	public final static int MOD = 1000000007;
+	public static long[][] m = {{1, 1}, {1, 0}};
+	public static long[][] m1 = {{1, 0}, {0, 1}};
```
Comment (SansPapyrus683) [inline content/5_Plat/Matrix_Expo.mdx:142]: what do these matrices represent?

File: content/5_Plat/Matrix_Expo.mdx [hunk 12]
```diff
@@ -129,6 +129,61 @@ int main() {
...
+		long[][] half = power(matrix, n / 2, MOD);
+		long[][] square = mul(half, half, MOD);
+		if (n % 2 == 0) {
+			return square;
+		}
+		return mul(matrix, square, MOD);
+	}
+	
+	public static void main(String[] args) {
+		Scanner sc = new Scanner(System.in);
```
Comment (SansPapyrus683) [inline content/5_Plat/Matrix_Expo.mdx:172]: don't use scanner

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Comment by SansPapyrus683** [comment]
hey, any updates on this pr?

File: content/5_Plat/Matrix_Expo.mdx (hunk 11 above)
Comment (19WH1A0524) [reply content/5_Plat/Matrix_Expo.mdx:142]: base matrix

File: content/5_Plat/Matrix_Expo.mdx (hunk 12 above)
Comment (19WH1A0524) [reply content/5_Plat/Matrix_Expo.mdx:172]: Should I use buffer reader or something?

File: content/5_Plat/Matrix_Expo.mdx (hunk 10 above)
Comment (ChuangSheep) [reply content/5_Plat/Matrix_Expo.mdx]: actually why `mod`? It is a constant

**Comment by 19WH1A0524** [comment]
I updated it

File: content/5_Plat/Matrix_Expo.mdx [hunk 13]
```diff
@@ -129,6 +129,60 @@ int main() {
...
 </CPPSection>
+<JavaSection>
+
+```java
+
+import java.util.*;
+import java.io.*;
+
+class FibonacciNumbers {
+	public final static int mod = 1000000007;
```
Comment (SansPapyrus683) [inline content/5_Plat/Matrix_Expo.mdx:140]: rename to `MOD`

File: content/5_Plat/Matrix_Expo.mdx (hunk 11 above)
Comment (SansPapyrus683) [reply content/5_Plat/Matrix_Expo.mdx:142]: the readers don't know that

File: content/5_Plat/Matrix_Expo.mdx [hunk 14]
```diff
@@ -129,6 +129,60 @@ int main() {
...
+	public static long[][] m = {{1, 1}, {1, 0}};
+	public static long[][] m1 = {{1, 0}, {0, 1}};
+
```
Comment (SansPapyrus683) [inline content/5_Plat/Matrix_Expo.mdx:143]:
```suggestion

	/** Multiplies the two matrices a and b. */
```

File: content/5_Plat/Matrix_Expo.mdx (hunk 12 above)
Comment (SansPapyrus683) [reply content/5_Plat/Matrix_Expo.mdx:172]: yes

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**